fabric-rendering-fluids-v1/src/client/java/net/fabricmc/fabric/mixin/client/rendering/fluid/FluidRendererMixin.java

```diff
@@ -76,58 +77,55 @@ public void onHeadRender(BlockRenderView view, BlockPos pos, VertexConsumer vert
-		// First boolean local is set by vanilla according to 'matches lava'
-		// but uses the negation consistent with 'matches water'
-		// for determining if overlay water sprite should be used behind glass.
-
-		// Has other uses but those are overridden by this mixin and have
-		// already happened by the time this hook is called.
-
-		// If this fluid has an overlay texture, set this boolean to false.
-		final FluidRenderHandlerInfo info = FluidRenderingImpl.getCurrentInfo();
-		return info.handler != null ? !info.hasOverlay : chk;
-	}
-
-	@ModifyVariable(at = @At(value = "INVOKE", target = "Lnet/minecraft/client/render/block/FluidRenderer;isSameFluid(Lnet/minecraft/fluid/FluidState;Lnet/minecraft/fluid/FluidState;)Z"), method = "render", ordinal = 0)
-	public Sprite[] modSpriteArray(Sprite[] chk) {
+	@ModifyVariable(
+			method = "render",
+			at = @At("STORE"),
+			ordinal = 0
+	)
+	public Sprite[] modSpriteArray(Sprite[] original) {
 		FluidRenderHandlerInfo info = FluidRenderingImpl.getCurrentInfo();
-		return info.handler != null ? info.sprites : chk;
+		return info.handler != null ? info.sprites : original;
 	}
 
-	@ModifyVariable(at = @At(value = "CONSTANT", args = "intValue=16", ordinal = 0, shift = At.Shift.BEFORE), method = "render", ordinal = 0)
-	public int modTintColor(int chk, BlockRenderView world, BlockPos pos, VertexConsumer vertexConsumer, BlockState blockState, FluidState fluidState) {
+	@ModifyVariable(method = "render", at = @At("STORE"), ordinal = 0)
+	public int modTintColor(int original, BlockRenderView world, BlockPos pos, VertexConsumer vertexConsumer, BlockState blockState, FluidState fluidState) {
 		FluidRenderHandlerInfo info = FluidRenderingImpl.getCurrentInfo();
-		return info.handler != null ? info.handler.getFluidColor(world, pos, fluidState) : chk;
+		return info.handler != null ? info.handler.getFluidColor(world, pos, fluidState) : original;
 	}
 
-	// Redirect redirects all 'waterOverlaySprite' gets in 'render' to this method, this is correct
-	@Redirect(at = @At(value = "FIELD", opcode = Opcodes.GETFIELD, target = "Lnet/minecraft/client/render/block/FluidRenderer;waterOverlaySprite:Lnet/minecraft/client/texture/Sprite;"), method = "render")
-	public Sprite modWaterOverlaySprite(FluidRenderer self) {
-		FluidRenderHandlerInfo info = FluidRenderingImpl.getCurrentInfo();
-		return info.handler != null && info.hasOverlay ? info.overlaySprite : waterOverlaySprite;
-	}
-
-	@Redirect(at = @At(value = "INVOKE", target = "Lnet/minecraft/block/BlockState;getBlock()Lnet/minecraft/block/Block;"), method = "render")
-	public Block getOverlayBlock(BlockState state) {
-		Block block = state.getBlock();
-		neighborBlock.set(block);
-
-		// An if-statement follows, we don't want this anymore and 'null' makes
-		// its condition always false (due to instanceof)
-		return null;
-	}
+	@Definition(id = "getFrameU", method = "Lnet/minecraft/client/texture/Sprite;getFrameU(F)F")
+	@Expression("@(?).getFrameU(0.0)")
```

Comment:
Don't target the wildcard here, use a concrete local. This ensures that the selected thing is actually targetable at all, which isn't a guarantee with a wildcard, but more importantly you *want* to ensure the receiver is the same local you're modifying